### PR TITLE
Added textfields to modify charbuild

### PR DESCRIPTION
**Pull request by Chen1571**

### CODE REVIEW

File: src/views/CharacterBuild.js [hunk 1]
```diff
@@ -22,12 +46,41 @@ export default function CharacterBuild() {
...
+                    color: "white"
+                  }
+                }}
+                size="small"
+                className={classes.field}
+                onChange={e => {
+                  if (stats === "character_name") {
+                    charBuild[stats] = e.target.value;
+                  } else {
+                    charBuild[stats] = parseInt(e.target.value);
```
Comment (dli7077) [inline src/views/CharacterBuild.js]: ParseFloat instead

File: src/views/CharacterBuild.js [hunk 2]
```diff
@@ -22,12 +46,41 @@ export default function CharacterBuild() {
...
+                variant="outlined"
+                inputProps={{
+                  style: {
+                    color: "white"
+                  }
+                }}
+                size="small"
+                className={classes.field}
+                onChange={e => {
+                  if (stats === "character_name") {
```
Comment (dli7077) [inline src/views/CharacterBuild.js]: Character name should not be allowed to be changed. Use _.omit to remove it from the stat list

File: src/views/CharacterBuild.js [hunk 3]
```diff
@@ -22,12 +46,41 @@ export default function CharacterBuild() {
...
+              marginRight="20px"
+            >
               {stats}
             </Box>
             <Box style={{ justifyContent: "flex-end", minWidth: "100px" }}>
-              {get(charBuild, stats)}
+              <TextField
+                type={stats === "character_name" ? "text" : "number"}
+                label={stats}
+                variant="outlined"
```
Comment (dli7077) [inline src/views/CharacterBuild.js]: Remove variant. The default with just line looks cleaner

File: src/views/CharacterBuild.js [hunk 4]
```diff
@@ -22,12 +46,41 @@ export default function CharacterBuild() {
...
+                inputProps={{
+                  style: {
+                    color: "white"
```
Comment (dli7077) [inline src/views/CharacterBuild.js]: Can we give it the same genshin font?

**Review by dli7077 — CHANGES_REQUESTED** [review]
Character name should not be modified

File: src/views/CharacterBuild.js (hunk 2 above)
Comment (Chen1571) [reply src/views/CharacterBuild.js]: do we just use a static character name for now then? or a drop down menu of the available ones

File: src/views/CharacterBuild.js (hunk 3 above)
Comment (Chen1571) [reply src/views/CharacterBuild.js]: the textbox is like invisible without an outline, unless u want it to be filled
or u can change the background color

Comment (dli7077) [reply src/views/CharacterBuild.js]: thats fine, just change the on hover color to white or smth
image

File: src/views/CharacterBuild.js (hunk 2 above)
Comment (dli7077) [reply src/views/CharacterBuild.js]: static name, don't worry about other characters. Just pretend this calc is just for chongyun

Comment (dli7077) [reply src/views/CharacterBuild.js]: once we complete the actual calc, then we can worry about adding other characters and tailoring to them

File: src/views/CharacterBuild.js [hunk 5]
```diff
@@ -22,12 +34,42 @@ export default function CharacterBuild() {
...
+              }}
+              marginRight="20px"
+            >
               {stats}
             </Box>
             <Box style={{ justifyContent: "flex-end", minWidth: "100px" }}>
-              {get(charBuild, stats)}
+              <TextField
+                type={stats === "character_name" ? "text" : "number"}
+                disabled={stats === "character_name" ? true : false}
```
Comment (dli7077) [inline src/views/CharacterBuild.js]: id prefer you make a separate component just for character

so map through
`omit(keys(charBuild), 'character_name')` instead

generally, we may have more properties that should not be modified, so we should separate it from the 'modifiables'

**Review by dli7077 — CHANGES_REQUESTED** [review]
move character to its own component

File: src/views/CharacterBuild.js (hunk 5 above)
Comment (dli7077) [reply src/views/CharacterBuild.js]: also, can we remove this padding?
I know we should worry about styling later, but the padding fills up the page and testing will be hard if we need to constantly scroll to see everything

image

Comment (dli7077) [reply src/views/CharacterBuild.js]: if not possible, then dw

Comment (Chen1571) [reply src/views/CharacterBuild.js]: for the separate component do u still want a textfield?
or can i make a static box that says "Chongyun" for now

Comment (Chen1571) [reply src/views/CharacterBuild.js]: image

Comment (dli7077) [reply src/views/CharacterBuild.js]: static box is fine

Comment (dli7077) [reply src/views/CharacterBuild.js]: dont hard code chongyun, make it display whatever the charBuild's character_name is

File: src/views/CharacterBuild.js [hunk 6]
```diff
@@ -62,13 +72,7 @@ export default function CharacterBuild() {
                 }}
                 size="small"
                 className={classes.field}
-                onChange={e => {
-                  if (stats === "character_name") {
-                    charBuild[stats] = e.target.value;
-                  } else {
-                    charBuild[stats] = parseFloat(e.target.value);
-                  }
-                }}
```
Comment (dli7077) [inline src/views/CharacterBuild.js]: good job cleaning it up, but we still need defaultValue to see the stats

Comment (Chen1571) [reply src/views/CharacterBuild.js]: what do you mean by needing it to see the stats

**Review by dli7077 — APPROVED** [review]
good job!

**Pull request merged**